src/main/haskell/kore/src/Kore/Unification/Procedure.hs

```diff
@@ -86,21 +89,25 @@ unificationProcedure tools substitutionSimplifier p1 p2
                 p1
                 p2
     (pat@Predicated { term, predicate, substitution }, _) <- getUnifiedTerm
-    let
-        (pred', _) = Ceil.makeEvaluateTerm tools term
     if Predicated.isBottom pat
         then return
             (OrOfExpandedPattern.make [], EmptyUnificationProof)
-        else return
-            ( OrOfExpandedPattern.make
-                [ Predicated
-                    { term = ()
-                    , predicate = makeAndPredicate predicate pred'
-                    , substitution
-                    }
-                ]
-            , EmptyUnificationProof
-            )
+        else Monad.Trans.lift $ do
```

Comment:
Is the `else` case here checking that the returned unifier is defined, by wrapping the term in `\ceil`? Why would `termUnification` ever return an undefined unifier that is not simply `\bottom`?